lib/services/update-factory.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const h = require('highland'),
+  pageList = require('../page-list');
+
+/**
+ * Test the request body for a url property.
+ *
+ * @param  {Object} body
+ * @return {Object}
+ */
+function errOrGood(body) {
+  if (!body || !body.url) {
+    let err = new Error('`_pagelist` endpoint cannot update a page without a url');
+
+    err.code = 400;
+    return err;
+  }
+
+  return body;
+}
+
+/**
+ * Format an error message
+ *
+ * @param  {Error} err
+ * @return {Object}
+ */
+function handleError({ message, code }) {
+  return { status: 'error', msg: message, code };
+}
+
+/**
+ * If Error, handle it, if all good, run a query.
+ *
+ * @param  {Error|Object} arg
+ * @return {Stream}
+ */
+function queryOrPass(arg) {
```

Comment:
It'd be simpler to make it the business of `responses.streamOperation` to catch errors and turn them into `{status, msg, code}` objs instead of polluting the stream of this service with errors and constantly checking for them.

Reply:
If this wants to be used more universally then the error handling needs to be done in the operation streams using it. 

A bad request to `_pagelist` could trigger a `400` error
Bad logic in querying could trigger a 500.

There are other error handling architectures that I want to look at, but I want to move ALL HTTP responses to streaming. Future conversions will see this updated.

Reply:
It can be wait for future changes, but: We can conduct error handling inside the stream operation without polluting the stream with errors. Catch, add appropriate `err.code`, and throw again. StreamOperation catches and sets response code based on `err.code`. Passing errors into the stream only forces us to repeatedly check whether we're dealing with errors or actual data.